src/App.tsx

```diff
@@ -8,7 +7,47 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [selectType, setSelectType] = useState('All');
+  const [input, setInput] = useState('');
+  const [todoId, setTodoId] = useState(0);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const data = await getTodos()
+        .then(response => {
+          setTodos(response);
+          setIsLoading(false);
+        });
+
+      return data;
+    };
+
+    fetchData();
+  }, []);
+
+  const visibleTodos = todos
+    .filter(todo => {
+      switch (selectType) {
+        case 'active':
+          return !todo.completed;
+
+        case 'completed':
+          return todo.completed;
+
+        default:
+          return true;
+      }
+    })
+    .filter(todo => {
+      return todo.title.toLowerCase().includes(input.toLowerCase());
+    });
```

Comment:
you can create method for check includes `checkContainsQueryInTitle` and use it on your filter:
```
const visibleTodos = todos
    .filter(todo => {
      switch (selectType) {
        case 'active':
          return !todo.completed && checkContainsQueryInTitle(todo.title);

        case 'completed':
          return todo.completed && checkContainsQueryInTitle(todo.title);

        default:
          return true && checkContainsQueryInTitle(todo.title);
      }
    })
```